test/token-locking.js

```diff
@@ -0,0 +1,179 @@
+      await token.approve(tokenLocking.address, usersTokens);
+      await tokenLocking.deposit(token.address, usersTokens / 2);
+      await tokenLocking.deposit(token.address, usersTokens / 2);
+      const userDepositedBalance = await tokenLocking.getUserLockedBalance.call(token.address, addresses[0]);
+      assert.equal(userDepositedBalance.toNumber(), usersTokens);
+    });
+
+    it("should not be able to deposit tokens while they are locked", async () => {
+      await token.approve(tokenLocking.address, usersTokens);
+      await tokenLocking.deposit(token.address, usersTokens);
+      await colony.startNextRewardPayout(otherToken.address);
+      await checkErrorRevert(tokenLocking.deposit(token.address, usersTokens));
+    });
+
+    it("should not be able to withdraw tokens while they are locked", async () => {
+      await token.approve(tokenLocking.address, usersTokens);
+      await tokenLocking.deposit(token.address, usersTokens);
+      await colony.startNextRewardPayout(otherToken.address);
+      await checkErrorRevert(tokenLocking.withdraw(token.address, usersTokens));
+    });
+
+    it("should be able to unlock tokens while they are locked", async () => {
+      await token.approve(tokenLocking.address, usersTokens);
+      await tokenLocking.deposit(token.address, usersTokens);
+      const { logs } = await colony.startNextRewardPayout(otherToken.address);
+      const payoutId = logs[0].args.id;
+      await colony.waiveRewardPayout(payoutId);
+      await tokenLocking.withdraw(token.address, usersTokens);
+      const userDepositedBalance = await tokenLocking.getUserLockedBalance.call(token.address, addresses[0]);
+      assert.equal(userDepositedBalance, 0);
+    });
+
+    it("should not be able to lock tokens twice", async () => {
+      await token.approve(tokenLocking.address, usersTokens);
+      await tokenLocking.deposit(token.address, usersTokens);
+      await colony.startNextRewardPayout(otherToken.address);
+
+      const tokenArgs = getTokenArgs();
+      const newToken = await Token.new(...tokenArgs);
+      await colony.startNextRewardPayout(newToken.address);
+
+      const totalLockCount = await tokenLocking.getTotalTokenLockCount.call(token.address);
+      assert.equal(totalLockCount.toNumber(), 2);
+    });
+
+    it("should change timestamp when user deposit their tokens", async () => {
+      await token.approve(tokenLocking.address, usersTokens);
+      let info = await tokenLocking.getUserTokenLockInfo.call(token.address, addresses[0]);
+      const lastDepositTimestampsBeforeDeposit = info[1];
+      await forwardTime(1000);
+      await tokenLocking.deposit(token.address, usersTokens);
+      info = await tokenLocking.getUserTokenLockInfo.call(token.address, addresses[0]);
+      const lastDepositTimestampsAfterDeposit = info[1];
+
+      assert(lastDepositTimestampsAfterDeposit > lastDepositTimestampsBeforeDeposit, "timestamp of deposit is not updated correctly");
```

Comment:
You should really be getting the blocktime and testing you are close to that. See `testHelper.currentBlockTime` and `assert.closeTo` in `mocha`.